src/components/oculus-touch-controls.js

```diff
@@ -320,24 +360,128 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
+    buttonRanges.trigger = {
+      min: controllerObject3D.getObjectByName('xr_standard_trigger_pressed_min'),
+      max: controllerObject3D.getObjectByName('xr_standard_trigger_pressed_max')
+    };
+    buttonRanges.trigger.diff = {
+      x: Math.abs(buttonRanges.trigger.max.rotation.x) - Math.abs(buttonRanges.trigger.min.rotation.x),
+      y: Math.abs(buttonRanges.trigger.max.rotation.y) - Math.abs(buttonRanges.trigger.min.rotation.y),
+      z: Math.abs(buttonRanges.trigger.max.rotation.z) - Math.abs(buttonRanges.trigger.min.rotation.z)
+    };
+
+    var button1 = this.data.hand === 'left' ? 'x' : 'a';
+    var button2 = this.data.hand === 'left' ? 'y' : 'b';
+    var button1id = button1 + 'button';
+    var button2id = button2 + 'button';
+
+    buttonMeshes[button1id] = controllerObject3D.getObjectByName(button1 + '_button');
+    buttonObjects[button1id] = controllerObject3D.getObjectByName(button1 + '_button_pressed_value');
+    buttonRanges[button1id] = {
+      min: controllerObject3D.getObjectByName(button1 + '_button_pressed_min'),
+      max: controllerObject3D.getObjectByName(button1 + '_button_pressed_max')
+    };
+
+    buttonMeshes[button2id] = controllerObject3D.getObjectByName(button2 + '_button');
+    buttonObjects[button2id] = controllerObject3D.getObjectByName(button2 + '_button_pressed_value');
+    buttonRanges[button2id] = {
+      min: controllerObject3D.getObjectByName(button2 + '_button_pressed_min'),
+      max: controllerObject3D.getObjectByName(button2 + '_button_pressed_max')
+    };
+
+    buttonRanges[button1id].unpressedY = buttonObjects[button1id].position.y;
+    buttonRanges[button1id].pressedY =
+      buttonRanges[button1id].unpressedY + Math.abs(buttonRanges[button1id].max.position.y) - Math.abs(buttonRanges[button1id].min.position.y);
+
+    buttonRanges[button2id].unpressedY = buttonObjects[button2id].position.y;
+    buttonRanges[button2id].pressedY =
+      buttonRanges[button2id].unpressedY - Math.abs(buttonRanges[button2id].max.position.y) + Math.abs(buttonRanges[button2id].min.position.y);
+  },
+
   onAxisMoved: function (evt) {
     emitIfAxesChanged(this, this.mapping[this.data.hand].axes, evt);
   },
 
+  onThumbstickMoved: function (evt) {
+    if (!this.isOculusTouchV3 || !this.buttonMeshes || !this.buttonMeshes.thumbstick) { return; }
+    for (var axis in evt.detail) {
+      this.buttonObjects.thumbstick.rotation[this.axisMap[axis]] =
+        this.buttonRanges.thumbstick.originalRotation[this.axisMap[axis]] -
+        (Math.PI / 8) *
+        evt.detail[axis] *
+        (axis === 'y' || this.data.hand === 'right' ? -1 : 1);
+    }
+  },
+  axisMap: {
+    y: 'x',
+    x: 'z'
+  },
+
   updateModel: function (buttonName, evtName) {
     if (!this.data.model) { return; }
     this.updateButtonModel(buttonName, evtName);
   },
 
-  updateButtonModel: function (buttonName, state) {
+  updateButtonModel: function (buttonName, evtName) {
```

Comment:
Well, this was a conscious change. If you'll humor me, this feels like a mistake.

### tldr:
- updateModel calls this argument evtName
- this is indeed only a name, and not state
- onButtonChanged follows a different flow and actually _does_ use `evt.detail.state`, which is unavailable here
- calling this state incorrectly feels distinctly like an oversight

### Detailed argument:
- oculus-touch-controls.js adds onButtonDown (etc.) event listeners, have them point to utils/tracked-controls.js/onButtonEvent()
- tracked-controls.js emits 'buttondown'
- oculus-touch-controls.js picks that up, and strips away the event and only passes the detail.id and the word 'down'; state is ignored
- utils/tracked-controls.js/onButtonEvent() calls oculus-touch-controls.js/updateModel() directly
- updateModel() calls the argument it receives `evtName` (correctly, as it does not include state)
- but then, updateModel() calls updateButtonModel(), which now calls that `state` instead of `evtName`

This is particularly confusing, because evt.detail.state actually is a thing that exists, but it isn't available here.

There is a different flow where state is made accessible:
- oculus-touch-controls.js adds onButtonChanged event listener
- this is different than the onButton Down / Up / TouchStart / TouchEnd events; it doesn't use that utils function, but instead passes the raw event, _which includes event.detail.state_, to its own function (_not_ the utils onButtonEvent() function)
- evt.detail.state.value is then used to precisely adjust the mesh movement

Based on all that, it seemed like an oversight that this argument was called state instead of evtName, being that both are clearly defined concepts in this file and outside of this file.

Reply:
It's better not to change anything unrelated to the PR. It makes reviews more complicated